### PR TITLE
Draft of the Echo Handler.

### PR DESCRIPTION
**Pull request by avikbmohan**

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
Request,Repsonse and handlers should be put into different files.

File: server/request_handler.cc [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#include 'request_handler.h'
+#include 'http.h'
+#include <string>
+
+EchoHandler::Status Echohandler::Init(const std::string& uri_prefix,
+				      const NginxConfig& config) {
+  /*  m_uri_prefix_ = uri_prefix;
```
Comment (richardmin) [inline server/request_handler.cc:7]: Why is this code commented out? Regardless, don't merge commented out code.

File: server/request_handler.h [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+  // Handles an HTTP request, and generates a response. Returns a response code
+  // indicating success or failure condition. If ResponseCode is not OK, the
+  // contents of the response object are undefined, and the server will return
+  // HTTP code 500.
+  virtual Status HandleRequest(const Request& request,
+                               Response* response) = 0;
+};
+
+class EchoHandler: public RequestHandler {
```
Comment (richardmin) [inline server/request_handler.h]: Make a separate file for this.

File: server/request_handler.h [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+ public:
+  enum Status {
```
Comment (richardmin) [inline server/request_handler.h]: You don't have to overwrite the enum.

File: server/request_handler.h [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    OK = 0,
+    BAD = -1
+  };
+  
+  Status Init(const std::string& uri_prefix,
```
Comment (richardmin) [inline server/request_handler.h]: I *believe* you don't have to redeclare functions if you're overriding them.

**Review by richardmin — CHANGES_REQUESTED** [review]
Add tests.

**Review by richardmin — CHANGES_REQUESTED** [review]
There are merge conflicts here?

It's very hard to understand what's going on right now.

**Comment by richardmin** [comment]
Out of date.

**Pull request closed without merging**